src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/UseNameofInPlaceOfString.cs

```diff
@@ -0,0 +1,152 @@
+        internal const string RuleId = "CA1507";
+        private const string ParamName = "paramName";
+        private const string PropertyName = "propertyName";
+        internal const string StringText = "StringText";
+
+        private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(MicrosoftMaintainabilityAnalyzersResources.UseNameOfInPlaceOfStringTitle), MicrosoftMaintainabilityAnalyzersResources.ResourceManager, typeof(MicrosoftMaintainabilityAnalyzersResources));
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(MicrosoftMaintainabilityAnalyzersResources.UseNameOfInPlaceOfStringMessage), MicrosoftMaintainabilityAnalyzersResources.ResourceManager, typeof(MicrosoftMaintainabilityAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftMaintainabilityAnalyzersResources.UseNameOfInPlaceOfStringDescription), MicrosoftMaintainabilityAnalyzersResources.ResourceManager, typeof(MicrosoftMaintainabilityAnalyzersResources));
+
+        internal static DiagnosticDescriptor RuleWithSuggestion = new DiagnosticDescriptor(RuleId,
+                                                                         s_localizableTitle,
+                                                                         s_localizableMessage,
+                                                                         DiagnosticCategory.Maintainability,
+                                                                         DiagnosticHelpers.DefaultDiagnosticSeverity,
+                                                                         isEnabledByDefault: true,
+                                                                         description: s_localizableDescription,
+                                                                         helpLinkUri: "https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.CodeQuality.Analyzers/Microsoft.CodeQuality.Analyzers.md#maintainability",
+                                                                         customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(RuleWithSuggestion);
+
+        public override void Initialize(AnalysisContext analysisContext)
+        {
+            analysisContext.EnableConcurrentExecution();
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            analysisContext.RegisterOperationAction(AnalyzeArgument, OperationKind.Argument);
+        }
+
+        private void AnalyzeArgument(OperationAnalysisContext context)
+        {
+            var argument = (IArgumentOperation)context.Operation;
+            if ((argument.Value.Kind != OperationKind.Literal 
+                || argument.Value.Type.SpecialType != SpecialType.System_String))
```

Comment:
`argument.Value.Type` could be null for a `null` literal. Can you add a null check and also a corresponding unit test?

Reply:
Passing a null literal as a parameter causes it to be converted to the type of the expected parameter so argument.Value.Type won't actually be null.  But it will have a different OperationKind so will still fail under the existing check.  I have added this as a unit test.

Reply:
is it still true for error case where Method signature contains error like missing type or error type? unit test?

Reply:
@heejaechang We no longer generate `IInvocationOperation` or `IArgumentOperation` for error cases, instead it is an `IInvalidOperation`. I think @chborl added tests for these.

Reply:
👍